tests/e2e/testcases/icm/sendMsg/suite.go

```diff
@@ -0,0 +1,522 @@
+			})
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			rpcs1, err := utils.ParseRPCsFromOutput(output)
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			// Deploy ICM to subnet2
+			output, err = commands.DeployICMContracts([]string{}, utils.TestFlags{
+				"key":        ewoqKeyName,
+				"blockchain": subnet2Name,
+			})
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			rpcs2, err := utils.ParseRPCsFromOutput(output)
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			// Deploy the relayer
+			_, err = commands.DeployRelayer(
+				[]string{
+					"deploy",
+					"--cchain",
+				},
+				utils.TestFlags{
+					"key":           ewoqKeyName,
+					"blockchains":   fmt.Sprintf("%s,%s", subnetName, subnet2Name),
+					"amount":        10000,
+					"cchain-amount": 10000,
+					"log-level":     "info",
+				})
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			// Send a message
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
+
+			sendMsgFlags := utils.TestFlags{
+				"key":        ewoqKeyName,
+				"source-rpc": rpcs2[0],
+				"dest-rpc":   rpcs1[0],
+			}
+
+			sendMessageArgs := []string{
+				subnet2Name,
+				subnetName,
+				message,
+			}
+
+			output, err = utils.TestCommand(utils.ICMCmd, "sendMsg", sendMessageArgs, globalFlags, sendMsgFlags)
+			gomega.Expect(err).Should(gomega.BeNil())
+			gomega.Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("Delivering message \"%s\" from source blockchain \"%s\"", message, subnet2Name)))
+			gomega.Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("Waiting for message to be delivered to destination blockchain \"%s\"", subnetName)))
+			gomega.Expect(output).Should(gomega.ContainSubstring("Message successfully Teleported!"))
+
+			commands.StopRelayer()
+		})
+
+		ginkgo.It("should transfer hex encoded message", func() {
+			commands.CreateSubnetEvmConfigNonSOV(subnetName, utils.SubnetEvmGenesisPath, false)
+			commands.DeploySubnetLocallyNonSOV(subnetName)
+
+			// Deploy ICM
+			_, err := commands.DeployICMContracts([]string{}, utils.TestFlags{
+				"key":        ewoqKeyName,
+				"blockchain": subnetName,
+			})
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			// Deploy the relayer
+			_, err = commands.DeployRelayer(
+				[]string{
+					"deploy",
+					"--cchain",
+				},
+				utils.TestFlags{
+					"key":           ewoqKeyName,
+					"blockchains":   subnetName,
+					"amount":        10000,
+					"cchain-amount": 10000,
+					"log-level":     "info",
+				})
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			// Send a message
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
+
+			sendMsgFlags := utils.TestFlags{
+				"key": ewoqKeyName,
+			}
+
+			hexEncodedMessage := "0x48656c6c6f20576f726c64" // "Hello World" in hex
+			sendMessageArgs := []string{
+				cChain,
+				subnetName,
+				hexEncodedMessage,
+				"--hex-encoded",
```

Comment:
@felipemadero any reason why we would ever need hex-encoded flag?

Reply:
to provide any byte data as a test message, not just strings. needed for more complex interactions with
contracts, eg to simulate contract calls

Reply:
what exactly are hex encoded flags? I will update the flag description so that its more easily understood by users @felipemadero

Reply:
I believe the user provides the string in a hex format and it is sent that way, so it's more flexible. Not sure if it's actually used much